akka-http-core/src/test/scala/akka/http/impl/engine/client/HostConnectionPoolSpec.scala

```diff
@@ -190,18 +190,45 @@ class HostConnectionPoolSpec extends AkkaSpecWithMaterializer(
 
         conn1.expectRequestToPath("/2")
       }
-      "not get stuck on HEAD requests if payload is never subscribed" inWithShutdown new SetupWithServerProbes(_.withResponseEntitySubscriptionTimeout(Duration.Inf).withMaxConnections(1)) {
-        pushRequest(HttpRequest(method = HttpMethods.HEAD, uri = "/head"))
-        pushRequest(HttpRequest(uri = "/2"))
-        val conn1 = expectNextConnection()
-        conn1.expectRequestToPath("/head")
+      if (clientServerImplementation != PassThrough) {
```

Comment:
Delicate, what? The test infra adds decorators to the entity in PassThrough mode, but no worries this is less realistic than the other tests.